src/track/serato/markers2.cpp

```diff
@@ -695,4 +711,20 @@ bool SeratoMarkers2::isBpmLocked() const {
     return false;
 }
 
+void SeratoMarkers2::setBpmLocked(bool bpmLocked) {
+    for (auto& pEntry : m_entries) {
+        DEBUG_ASSERT(pEntry);
+        if (pEntry->typeId() != SeratoMarkers2Entry::TypeId::BpmLock) {
+            continue;
+        }
+        SeratoMarkers2BpmLockEntry* pBpmLockEntry =
+                static_cast<SeratoMarkers2BpmLockEntry*>(pEntry.get());
+        pBpmLockEntry->setLocked(bpmLocked);
+        return;
```

Comment:
Add the assertion right here before the return statement. This is where it matters.